consensus/polybft/fsm.go

```diff
@@ -161,8 +161,31 @@ func (f *fsm) BuildProposal() (*pbft.Proposal, error) {
 	}
 
 	// fill the block with transactions
-	if err := f.blockBuilder.Fill(); err != nil {
-		return nil, err
+	receipts := f.blockBuilder.Fill()
+	events := make([]*ExitEvent, 0)
+
+	for i := 0; i < len(receipts); i++ {
+		if len(receipts[i].Logs) == 0 {
+			continue
+		}
+
+		event, err := decodeExitEvent(convert(receipts[i].Logs[0]), f.epochNumber, parent.Number+1)
```

Comment:
Not sure about receipts[i].Logs[0]. I think we need to check all logs.
cc @QEDK

Reply:
One more thing. If log.Event is not exit contract event than we can skip other computations immideatly.

Reply:
OK, I added checking of all logs in receipt.

Fix: https://github.com/0xPolygon/polygon-edge/pull/852/commits/d9bbc65d412e67074c623708a184a265a2fbbd4c